monai/losses/cldice.py

```
@@ -121,64 +108,245 @@ class SoftclDiceLoss(_Loss):
+            ValueError: When more than 1 of [``sigmoid=True``, ``softmax=True``, ``other_act is not None``].
+                Incompatible values.
+
         """
-        super().__init__()
+        super().__init__(reduction=LossReduction(reduction).value)
+        if other_act is not None and not callable(other_act):
+            raise TypeError(f"other_act must be None or callable but is {type(other_act).__name__}.")
+        if int(sigmoid) + int(softmax) + int(other_act is not None) > 1:
+            raise ValueError("Incompatible values: more than 1 of [sigmoid=True, softmax=True, other_act is not None].")
+        if smooth <= 0:
+            raise ValueError(f"smooth must be a positive value but got {smooth}.")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)